### PR TITLE
Adding SDKMan and Spotless to have a better dev experience

### PR DESCRIPTION
**Pull request by artur-ciocanu**
This tries to address a few "annoyances":
- SDKMan - to have proper version of Java
- Checkstyle from Dapr Java SDK - to ensure we have the same style

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -85,6 +86,38 @@
             <scope>test</scope>
         </dependency>
 	</dependencies>
+
```
Comment (salaboy) [inline pom.xml:89]: @artur-ciocanu this is quite good. but we probably need to follow the Dapr Java SDK formatting stuff..

File: .sdkmanrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Enable auto-env through the sdkman_auto_env config
```
[the remark this answers is not in the document]
Comment (artur-ciocanu) [reply .sdkmanrc:1]: Yeah, the SDKMan was copied from Java SDK.

File: pom.xml (hunk 1 above)
Comment (artur-ciocanu) [reply pom.xml:89]: I have used the Dapr Java SDK checkstyle.xml.

**Comment by artur-ciocanu** [comment]
@salaboy I have imported the `checkstyle.xml` from Dapr Java SDK. I have run the checkstyle and fixed all the issues, both in src/main and src/test.

Please take another look.

**Pull request merged**